src/memjs/server.ts

```diff
@@ -214,15 +214,16 @@ export class Server extends events.EventEmitter {
 
       // setup error handler
       self._socket.on("error", function (error) {
+        self.error(error);
+      });
+
+      self._socket.on("close", function () {
```

Comment:
This PR changes the ordering here. `on("error"` runs before `on("close"`. From reading the error handler, it seems like everything will be fine.

I do think we should terminate the currently pending requests in the case you talk about, where the socket closed but there's no error. We should be sure to tell every pending request that it errored because the socket closed:

https://github.com/makenotion/memjs/blob/53415596b3f54d57dddd0f7ef69914420c31046f/src/memjs/server.ts#L130-L132

by calling`self.error(new Error("Socket closed"))` just to be sure that pending requests are totally flushed in this case

Reply:
I tried this, seems not working as expected. Some sockets will close automatically after `idle`.
I also checked with memjs current repo's code. They also do not throw error when `on("close"`
So I will keep current code.

Reply:
ok!